temporallib/auth/auth.py

```diff
@@ -13,77 +13,50 @@
 from google.oauth2 import service_account
 import google.auth.transport.requests
 from typing import Union
-from dataclasses import asdict
-import os
+from pydantic_settings import BaseSettings
+from pydantic import Field, BaseModel
 
 
-class MacaroonAuthOptions:
-    """
-    Defines the parameters for authenticating with Candid.
-    """
+class MacaroonAuthOptions(BaseSettings):
+    macaroon_url: str
```

Comment:
FYI: This looks different from the environment variable you used before, `TEMPORAL_CANDID_MACAROON_URL` instead of `TEMPORAL_CANDID_URL`.

E: I see that you changed the tests to match. I don't know if you need to update any documentation using this.

Reply:
Great catch, thanks for pointing it out. I'm matching these with what's in the Temporal worker charm.